src/EditTableDialog.cpp

```diff
@@ -193,6 +200,7 @@ void EditTableDialog::updateColumnWidth()
     ui->tableConstraints->setColumnWidth(kConstraintColumns, 180);
     ui->tableConstraints->setColumnWidth(kConstraintType, 130);
     ui->tableConstraints->setColumnWidth(kConstraintName, 130);
+    ui->tableConstraints->setColumnWidth(kConstraintReference, 180);
```

Comment:
You're right that 180 is too small for the foreign key editor widget and that larger sizes would be odd in the dialog. I'm not sure either. You could try to make a new widget (and you might need to do that anyway in order to support other constraint types) which only contains a foreign key editor widget and a tool button and which makes sure the tool button is always visible. Clicking the tool button would then expand the widget into a popup window or so. I'm not sure though if that's easy to implement and/or very user friendly.